### PR TITLE
Create kotlin_jvm_image rule

### PR DESCRIPTION
**Pull request by alde**
proposed resolution of #301

heavily based on groovy_image rule

note: I'm still a bazel newbie and not entirely familiar with it, there are some unrelated tests (new_pusher_tests) that fail on master and on my branch.

### CODE REVIEW

File: kotlin/image.bzl [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    if srcs:
+        kt_jvm_library(
+            name = binary_name + "-lib",
+            srcs = srcs,
+            deps = deps + layers,
+        )
+        deps = deps + [binary_name + "-lib"]
+
+    # This always belongs in a separate layer.
+    # layers = layers + ["//external:kotlin"]
```
Comment (smukherj1) [inline kotlin/image.bzl]: Can this line be removed if not required?

**Comment by alde** [comment]
Thanks for taking a look @smukherj1. I think I've fixed the issue (I had the wrong test data).

I can build and run the image locally:
```bash
$ bazel run //tests/container/kotlin:kotlin_image_test
DEBUG: Rule 'ubuntu1604' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "eb50020790e22538676e17c0242b9272bdb5c81f7bc6b128a4abfa7ad31faf5b"
DEBUG: Call stack for the definition of repository 'ubuntu1604' which is a http_archive (rule definition at /home/dybeck/.cache/bazel/_bazel_dybeck/9a1a8dc804f6dd9cc5a0c132abdf646a/external/bazel_tools/tools/build_defs/repo/http.bzl:292:16):
 - /home/dybeck/git/private/rules_docker/WORKSPACE:233:1
DEBUG: Rule 'debian9' indicated that a canonical reproducible form can be obtained by modifying arguments sha256 = "eb50020790e22538676e17c0242b9272bdb5c81f7bc6b128a4abfa7ad31faf5b"
DEBUG: Call stack for the definition of repository 'debian9' which is a http_archive (rule definition at /home/dybeck/.cache/bazel/_bazel_dybeck/9a1a8dc804f6dd9cc5a0c132abdf646a/external/bazel_tools/tools/build_defs/repo/http.bzl:292:16):
 - /home/dybeck/git/private/rules_docker/WORKSPACE:239:1
INFO: Build option --cache_test_results has changed, discarding analysis cache.
INFO: Analyzed target //tests/container/kotlin:kotlin_image_test (0 packages loaded, 7276 targets configured).
INFO: Found 1 target...
Target //tests/container/kotlin:kotlin_image_test up-to-date:
  bazel-bin/tests/container/kotlin/kotlin_image_test
INFO: Elapsed time: 0.452s, Critical Path: 0.01s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //tests/container/kotlin:kotlin_image_test
-----------------------------------------------------------------------------
+ tests/container/kotlin/kotlin_image_test.image --norun
Loaded image ID: sha256:44bdc845143e4bf824832980bd722da3db4ef0bf6b2fb700bc381bf790fe636c
Tagging 44bdc845143e4bf824832980bd722da3db4ef0bf6b2fb700bc381bf790fe636c as testscontainerkotlinkt_jvm_image:intermediate
+ ../structure_test_linux/file/downloaded version
v1.8.0
+ ../structure_test_linux/file/downloaded test --driver docker --image testscontainerkotlinkt_jvm_image:intermediate --config tests/container/kotlin/configs/kt_jvm_image.yaml

==========================================
====== Test file: kt_jvm_image.yaml ======
==========================================
=== RUN: Metadata Test
--- PASS


===========================================
================= RESULTS =================
===========================================
Passes:      1
Failures:    0
Total tests: 1

PASS
```

and

```bash
$ docker run testscontainerkotlinkt_jvm_image:intermediate
Hello World
```

**Comment by alde** [comment]
sure, I could look into doing that

**Comment by alde** [comment]
I'm also going a bit back and forth on if it should be called `kt_jvm_image` (to follow `kt_jvm_binary`) or `kotlin_jvm_image` which I personally think is a more and easier to discover.

**Comment by nlopezgi** [comment]
> I'm also going a bit back and forth on if it should be called `kt_jvm_image` (to follow `kt_jvm_binary`) or `kotlin_jvm_image` which I personally think is a more and easier to discover.

I think `kt_jvm_image` is the right one as it follows `kt_jvm_binary`

**Comment by alde** [comment]
Thanks for the merge!

**Pull request merged**